src/Button/index.spec.js

```diff
@@ -0,0 +1,204 @@
+					<Button icon="ok" id="ok">
+						Ok
+					</Button>
+					<Button id="nope">Nope</Button>
+				</div>
+			</Wrapper>
+		);
+
+		const ok = container.querySelector('#ok');
+		const svg = ok.querySelector('svg');
+		const nope = container.querySelector('#nope');
+		const inexistent = nope.querySelector('svg');
+
+		expect(svg).toBeDefined();
+		expect(inexistent).toBeNull();
+	});
+
+	it('can render buttons with text and icons on the right side', () => {
+		const {container} = render(
+			<Wrapper>
+				<div>
+					<Button icon="ok" id="right" iconRight>
+						Ok
+					</Button>
+					<Button icon="ok" id="regular">
+						Ok
+					</Button>
+				</div>
+			</Wrapper>
+		);
+
+		const right = container.querySelector('#right');
+		const regular = container.querySelector('#regular');
+		const rightSvg = right.querySelector('svg');
+		const svg = regular.querySelector('svg');
+
+		const rightParent = rightSvg.parentElement;
+		const parent = svg.parentElement;
+
+		expect(rightParent.children[0]).not.toBe(rightSvg);
+		expect(parent.children[0]).toBe(svg);
+	});
+
+	it('can render buttons with subtitles', () => {
+		const {getByText} = render(
+			<Wrapper>
+				<Button id="ok" icon="ok" subline="subline">
+					Ok
+				</Button>
+			</Wrapper>
+		);
+		const text = getByText('Ok');
+		const subline = getByText('subline');
+		const parent = text.parentElement;
+		const {children} = parent;
+
+		const array = [...children];
+		const ixText = array.indexOf(text);
+		const ixSubline = array.indexOf(subline);
+
+		expect(ixText).toBeGreaterThan(-1);
+		expect(ixSubline).toBeGreaterThan(-1);
+		expect(ixText).toBeLessThan(ixSubline);
+	});
+
+	it('can render icon only buttons', () => {
+		const {container} = render(
+			<Wrapper>
+				<Button id="ok" icon="ok" />
+			</Wrapper>
+		);
+
+		const ok = container.querySelector('#ok');
+		const svg = ok.querySelector('svg');
+
+		expect(svg).toBeDefined();
+	});
+
+	it('can render round buttons', () => {
+		const {container} = render(
+			<Wrapper>
+				<div>
+					<Button id="round" round icon="plus" />
+					<Button id="regular" />
+				</div>
+			</Wrapper>
+		);
+
+		const round = container.querySelector('#round');
+		const regular = container.querySelector('#regular');
+
+		expect(round).toHaveStyle('border-radius: 2rem;');
+		expect(regular).not.toHaveStyle('border-radius: 2rem;');
+	});
+
+	it('can render large, medium and small sized buttons', () => {
+		const {container} = render(
+			<Wrapper>
+				<div>
+					<Button id="regular" />
+					<Button id="large" size="large" />
+					<Button id="medium" size="medium" />
+					<Button id="small" size="small" />
+				</div>
+			</Wrapper>
+		);
+
+		const regular = container.querySelector('#regular');
+		const large = container.querySelector('#large');
+		const medium = container.querySelector('#medium');
+		const small = container.querySelector('#small');
+		const {globals} = theme;
+
+		expect(regular).toHaveStyle(`
+			padding: ${pxToRem(12)}
```

Comment:
@diondiondion Could we add a reference property in the theme for this?

Reply:
@mistersomebody What do you mean?

Reply:
I think we should have a defined value in the theme for the button padding values. 

At the moment they are hard coded in the component which is not consistent with what is done for the font-size.

The test should be
```javascript
expect(regular).toHaveStyle(`
	padding: ${pxToRem(globals.padding.m)}
	font-size: ${globals.typeScale.m}
`);
```
Rather than:
```javascript
expect(regular).toHaveStyle(`
	padding: ${pxToRem(12)}
	font-size: ${globals.typeScale.m}
`);
```

Reply:
I see. It's a fair observation, but I'd rather keep the button's padding values outside of the theme, reason being that they're pretty arbitrary.

The spacing values in the theme are meant to provide consistency in layout & for spacing between components, not so much for "micro spacing" if that makes sense. For smaller components like buttons or form controls, finer control is needed to make things look nice, but adding every needed padding value to the theme would allow for too much variation in layout spacing & eventually defeat the point of it. It's not the most consistent approach, but more pragmatic. :)